### DISCUSSION TITLE
Router firmware only for Zigbee2mqtt or also ZHA

### DISCUSSION THREAD

**Discussion by Arthur-de-Partuur** [Koenkk/Z-Stack-firmware#451] [General]
Hi All,
Just curious if this router firmware is also suitable for ZHA intergration in Home Assistant or only for the Zigbee2MQTT integration?
If also for the ZHA, could anybody guide me to a method how to integrate the router into the system (i.e. putting it into bind mode?)

All help is appreciated!

**Comment by danpeig** [comment]
Yes. It works perfectly with ZHA.

When you flash the firmware it should be ready to pair as soon as you plug but if that does not work, open the dongle and press one of the buttons to enable paring mode.
